features/graph/api/src/main/java/org/opennms/netmgt/graph/api/generic/GenericElement.java

```diff
@@ -63,23 +62,15 @@ public void setProperty(String key, Object value) {
         return (T) properties.getOrDefault(key, defaultValue);
     }
 
-    public void setProperties(Map<String, Object> properties) {
-        this.properties.clear();
-        this.properties.putAll(properties);
-    }
-
     public Map<String, Object> getProperties() {
         return properties;
     }
 
+    // TODO: patrick: discuss with mvr if we don't want to have the namespace as a dedicated attribute in graph, vertex, edge
```

Comment:
@mvrueden , let's discuss

Reply:
The idea behind `Generic*` is that it has no dedicated attributes instead everything is persisted in a key/value-map.  What would be the benefit of introducing a `namespace` attribute? And if a `namespace` attribute is added, why not add an `id` attribute as well?

Reply:
it would make it a bit easier to set up the constructor chain, but agree we don't gain much by that. - let's leave it as a property for now.